CRM/EventsExtras/Hook/BuildForm.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use CRM_EventsExtras_ExtensionUtil as E;
+use CRM_EventsExtras_SettingsManager as SettingsManager;
+
+/**
+ * Abstract class for BuildForm Hook
+ */
+abstract class CRM_EventsExtras_Hook_BuildForm {
+
+  /**
+   * Event tab to display on the form 
+   */
+  protected $eventTab;
+
+  /**
+   * Constractor for BuildForm class
+   * 
+   * @param string $eventTab
+   * 
+   */
+  protected function __construct($eventTab){
+    $this->eventTab = $eventTab;
+    $templatePath = E::path() . '/templates/CRM/EventsExtras/Form/EventTabs.tpl';
+    CRM_Core_Region::instance('page-body')->add([
+      'template' => "{$templatePath}"
+    ]);
+  }
+
+  /**
+   * fuction to hide fields based on settings
+   *
+   * @param array $$form
+   *
+   */
+  protected function hideField(CRM_Event_Form_ManageEvent &$form){
+    $configFields = SettingsManager::getConfigFields();
```

Comment:
I think we can modify `SettingsManager::getConfigFields()` to optionally accept a section parameter. If we pass that in, it will return the fields relevant to the section passed in. That way we can encapsulate the logic in the SettingsManager class and if we change the way the sections are added to the fields later, we won't need to come here to change that.